tests/Entity/TransformerTest.php

```diff
@@ -0,0 +1,249 @@
+    {
+        $method = method_exists($this->pageManager, 'createWithDefaults') ? 'createWithDefaults' : 'create';
+        $this->pageManager->method($method)->willReturn(new SonataPagePage());
+        $this->pageManager->method('getClass')->willReturn(SonataPagePage::class);
+
+        $dateTime = new \DateTime();
+        $snapshot = new SonataPageSnapshot();
+        $snapshot->setContent($this->getTestContent($dateTime));
+        $snapshot->setUrl('/get-child');
+        $page = $this->transformer->load($snapshot);
+
+        static::assertSame('page_child', $page->getId());
+        static::assertSame('Page Child', $page->getName());
+        static::assertSame('Page Child Title', $page->getTitle());
+        static::assertSame('/get-child', $page->getUrl());
+    }
+
+    public function testLoadBlock(): void
+    {
+        $this->blockManager->method('create')->willReturnCallback(static fn () => new SonataPageBlock());
+
+        $dateTime = new \DateTime();
+
+        $page = new SonataPagePage();
+
+        $block = $this->transformer->loadBlock($this->getTestBlockArray($dateTime), $page);
+
+        static::assertSame('block123', $block->getId());
+    }
+
+    /**
+     * @phpstan-return PageContent
+     */
+    protected function getTestContent(\DateTimeInterface $datetime): array
+    {
+        return [
+            'id' => 'page_child',
+            'name' => 'Page Child',
+            'javascript' => null,
+            'stylesheet' => null,
+            'raw_headers' => null,
+            'title' => 'Page Child Title',
+            'meta_description' => null,
+            'meta_keyword' => null,
+            'template_code' => null,
+            'request_method' => 'GET|POST|HEAD|DELETE|PUT',
+            'created_at' => $datetime->format('U'),
+            'updated_at' => $datetime->format('U'),
+            'slug' => null,
+            'parent_id' => 'page_parent',
+            'target_id' => null, // REMOVE NEXT_MAYOR
+            'blocks' => [
+                $this->getTestBlockArray($datetime),
+            ],
+        ];
+    }
+
+    /**
+     * @phpstan-return BlockContent
+     */
+    protected function getTestBlockArray(\DateTimeInterface $datetime): array
+    {
+        return [
+            'id' => 'block123',
+            'name' => 'block1',
+            'enabled' => false,
+            'position' => 0,
+            'settings' => [],
+            'type' => 'type',
+            'created_at' => $datetime->format('U'),
+            'updated_at' => $datetime->format('U'),
+            'blocks' => [
+                [
+                    'id' => 'block234',
+                    'name' => 'block2',
+                    'enabled' => false,
+                    'position' => 0,
+                    'settings' => [],
+                    'type' => 'type',
+                    'created_at' => $datetime->format('U'),
+                    'updated_at' => $datetime->format('U'),
+                    'blocks' => [],
```

Comment:
are you covering this in other test, because I can't see!

and i't important, because it's the blocks inside the containers block for exmaple!

- header block container
   - menu block 1
   - menu block 2
- content block container
    - text block 1
    - text block 2

Reply:
you can see the json example here: https://github.com/sonata-project/SonataPageBundle/pull/1501#issuecomment-1196325158

Reply:
the current test data is Parent Page -> Child Page
The Child Page has 2 Blocks, one of them if the child of the other

but i can try your test data too

Reply:
> The Child Page has 2 Blocks, one of them if the child of the other

where are you checking this?

Reply:
block2 is the child of block1

Reply:
Yeah I saw this, but you are not checking the output (array), you need to check if the block2 is converted as expected! because for now you are expecting the the children be empty `[]`

Reply:
it should be like

```
'blocks' => [
                [
                    'id' => 'block234',
                    'name' => 'block1',
                    'enabled' => false,
                    'position' => 0,
                    'settings' => [],
                    'type' => 'type',
                    'created_at' => $datetime->format('U'),
                    'updated_at' => $datetime->format('U'),
                    'blocks' => [
                         [
                               'id' => 'block456',
                               'name' => 'block2',
                               'enabled' => false,
                         ],
                    ],
```

Reply:
the problem is you read this wrong,
we are already in the page blocks, from Line214-216

https://github.com/sonata-project/SonataPageBundle/pull/1516/files#diff-a14815f04837102213a846d666a6c4397b67e75309b476ebdd646dfc298bfd73R214-R216

so Line226+ is the first Block,
and Line236 is its child

Reply:
hmmm I got it! but just remember that we can have many levels of blocks, I don't know if it's usefull atm, but for now it's possible!

something like

- container 1
   - container 1.2
       .....
       - some other block 1